### ISSUE TITLE
Check the call of my_var(-1)

### ISSUE THREAD

**Issue by zchmielewska** [acturtle/cashflower#350]
Check what happens if the model calls `my_var(-1)`.

In Python, the negative index returns values from the list counting backwards. Results of a variable have a form of a list/array so it might be the same case.
```
my_lst = [1, 2, 3]
print(my_lst[-1])
# prints 3
```

To solve it, we need to unfortunately check it or solve it knowing the calculation direction.
For example:
```
# Negative index represents counting backward from the end, so it doesn't raise any error
if t < 0:
    raise IndexError("")
```

**Comment by zchmielewska** [comment]
A situation where this can happen is:

```
@variable()
def my_var(t):
    return my_var(t-1)
```

so in fact for `t=0` there should be an error because there is no `my_var(-1)` but the package does not throw any error currently.

This is because results are initialized with `np.empty` which means that the results are filled with whatever is in the memory and for `my_var(-1)` the last value from results is taken.

I think we have two options:
1. Add an if-statement checking if `t>=0` but that's a bummer in terms of speed.
2. Initialize results with `np.zeros`, then at least we have more sensible results (always zeros rather than some trash from the memory) - `np.zeros` is also slower than `np.empty` but better than if-statement.

**Comment by zchmielewska** [comment]
Implemented in 7fc9c19809fd721ea7e64b3e1edcb35dc1579cd1
